internal/mining/mining.go

```diff
@@ -1086,6 +1206,14 @@ func (g *BlkTmplGenerator) NewBlockTemplate(payToAddress dcrutil.Address) (*Bloc
 				}
 			}
 		}
+
+		// Obtain the maximum allowed treasury expenditure.
+		if isTreasuryEnabled {
+			maxTreasurySpend, err = g.chain.MaxTreasuryExpenditure(&prevHash)
```

Comment:
The `MaxTreasuryExpenditure` claims it can only be called on a block before a TVI, yet this seems to be to doing it on every block regardless.

Along these lines, I really dislike exported functions having these types of special "only call me at height foo" stuff attached to them, and there a lot of them that do that.  IMO, they should just return sane values when called in all cases, rather than trying to force easy to forget, and hard to get right, as this case shows, conditions on the caller. 

So, for example, I would expect `MaxTreasuryExpendeniture` to just return 0 with no error if called when spending from the treasury in the block after the one provided is not allowed for whatever reason.

Reply:
Fixed.